packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -138,6 +138,37 @@ export class ReadAttributeRequestItemProcessor extends GenericRequestItemProcess
                         )
                     );
                 }
+
+                const queryForPendingRelationship: any = {
+                    "peer.address": foundLocalAttribute.shareInfo.peer.address,
+                    status: RelationshipStatus.Pending
+                };
+                const pendingRelationshipToPeer = await this.accountController.relationships.getRelationships(queryForPendingRelationship);
+
+                const queryForActivatedOrTerminatedRelationship: any = {
+                    "peer.address": foundLocalAttribute.shareInfo.peer.address,
+                    $or: [
+                        {
+                            ["status"]: { $eq: RelationshipStatus.Active }
+                        },
+                        {
+                            ["status"]: { $eq: RelationshipStatus.Terminated }
+                        },
+                        {
+                            ["status"]: { $eq: RelationshipStatus.DeletionProposed }
+                        }
+                    ]
+                };
+
+                const activatedOrTerminatedRelationshipToPeer = await this.accountController.relationships.getRelationships(queryForActivatedOrTerminatedRelationship);
+
+                if (pendingRelationshipToPeer.length !== 0 || activatedOrTerminatedRelationshipToPeer.length === 0) {
```

Comment:
Do you mean that in the case that both a pending Relationship and a deletionProposed Relationship exist, the sharing of RelationshipAttributes is prevented, that actually belong to the deletionProposed Relationship and not to the pending Relationship and should therefore be able to be shared? 

If you are referring to the name of the variable `activatedOrTerminatedRelationshipToPeer`, I have interpreted deletion proposed Relationships as terminated Relationships (see coding) so as not to extend the variable name any further. But I can also change this if you have a better suggestion.

I also thought about adding a comment here, but since it does not comply with our code guidelines, I decided against it. I will think again about whether it wouldn't be better to add a comment here temporary. As soon as the change you mentioned is made to release/v5, the check ‘IF (activatedOrTerminatedRelationshipToPeer.length === 0)’ (currently also includes deletionProposed Relationships despite the naming) should be sufficient in terms of implementation and also understandable enough without comment.

Reply:
> Do you mean that in the case that both a pending Relationship and a deletionProposed Relationship exist, the sharing of RelationshipAttributes is prevented, that actually belong to the deletionProposed Relationship and not to the pending Relationship and should therefore be able to be shared?

This or the case you mentioned that the pending relationship has not been created yet which would make the validation successful.

To the comment: My only point was that you have to know about the problem with the order of creating the request and the relationship to understand the second check

Reply:
I agree. I find the way it is right now confusing.

Reply:
After the PR https://github.com/nmshd/runtime/pull/227 is merged, a Relationship in status `"Pending"` and a Relationship in status `"DeletionProposed"` can no longer exist at the same time, which is why the implementation can be simplified here and therefore no additional comment is required to explain it anymore.